setup.py

```diff
@@ -12,5 +12,5 @@
     packages=find_packages(),
     use_scm_version=True,
     setup_requires=['setuptools_scm'],
-    install_requires=open("requirements.txt").read().split(),
+    install_requires=open("requirements.txt").read().splitlines(),
```

Comment:
this fixes some of the issues that we had in requirements file (it was splitting on whitespace, not newline). We could also consider removing comments:

    [line for line in open("requirements.txt").read().splitlines() if line and not line.startswith('#')]

Still doesn't remove inline comments. I wonder if a specific [requirements parser](https://stackoverflow.com/a/59971236) would be a better idea?

Reply:
This fix with a comment in requirements.txt is good enough IMO. 

That said, I guess its fine to use a special parser for this or moving this other way around as that answer suggests etc. Maybe open a ticket for later improvement.

Reply:
of course, perhaps the better solution would be to switch to poetry/pyproject.toml instead, which should be able to reuse the same dependency list...